client/blocks/reader-join-conversation/dialog.jsx

```diff
@@ -10,20 +10,35 @@ import './style.scss';
 const ReaderJoinConversationDialog = ( { onClose, isVisible, loggedInAction, onLoginSuccess } ) => {
 	const translate = useTranslate();
 
-	const { login, createAccount } = useLoginWindow( { onLoginSuccess: onLoginSuccess } );
-
 	const trackEvent = ( eventName ) => {
 		let eventProps = {};
 		if ( loggedInAction ) {
-			eventProps = {
-				type: loggedInAction?.type,
-				blog_id: loggedInAction?.siteId,
-				post_id: loggedInAction?.postId,
-			};
+			eventProps = { type: loggedInAction?.type };
+			// create eventProps from loggedInAction properties if they exist
+			if ( loggedInAction?.siteId ) {
+				eventProps.blog_id = loggedInAction?.siteId;
+			}
+			if ( loggedInAction?.postId ) {
+				eventProps.post_id = loggedInAction?.postId;
+			}
+			if ( loggedInAction?.tag ) {
+				eventProps.tag = loggedInAction?.tag;
+			}
 		}
 		recordTracksEvent( eventName, eventProps );
 	};
 
+	const handleLoginSuccess = () => {
+		trackEvent( 'calypso_reader_dialog_login_success' );
+		onLoginSuccess();
+	};
+
+	if ( isVisible ) {
+		trackEvent( 'calypso_reader_action_clicked_requires_login' );
+	}
```

Comment:
If the component updates while `isVisible` is still true we risk sending this event in duplicates i think? So user does a requires_login action and this component renders `isVisible` and sends the event, then if another prop/parent updates and `isVisible` is still true we send the event again.

Alternatively I think we could put this in a `useEffect` and only send the event when `prevProps.isVisible === false && props.isVisible` or something similar.

Reply:
I was concerned something like that would happen but it didn't happen when testing. I'll look into the `useEffect` just in case though.